block/manager.go

```diff
@@ -261,9 +269,37 @@ func NewManager(
 		metrics:       seqMetrics,
 		isProposer:    isProposer,
 	}
+	agg.init(context.Background())
 	return agg, nil
 }
 
+func (m *Manager) init(ctx context.Context) {
+	// initialize da included height
+	height, err := m.store.GetMetadata(ctx, DAIncludedHeightKey)
+	if err == nil && len(height) == 8 {
+		m.daIncludedHeight.Store(binary.BigEndian.Uint64(height))
+	}
+}
+
+func (m *Manager) setDAIncludedHeight(ctx context.Context, newHeight uint64) error {
+	currentHeight := m.daIncludedHeight.Load()
```

Comment:
i think its better to make `Load()` and `Store()` concurrency bugproof like `SetHeight` in `store.go`. also, do you think its better to move the whole `daIncludedHeight` and corresponding functions to store.go?

Reply:
Using `CompareAndSwapNow`. Not sure about moving to store, will follow up later if it's not blocking.

Reply:
follow up issue: https://github.com/rollkit/rollkit/issues/1780